tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Process/ProcessStartCommon.cs

```diff
@@ -67,9 +72,220 @@ internal static Scope CreateScope(string filename, IDictionary<string, string> e
             return scope;
         }
 
-        internal static string Truncate(string value, int maxLength)
+        /// <summary>
+        /// Arguments > ArgumentList : If Arguments is used, ArgumentList is ignored. ArgumentsList is only used if Arguments is an empty string.
+        /// </summary>
+        private static ProcessCommandStartTags PopulateTags(string filename, IDictionary<string, string?>? environmentVariables, bool useShellExecute, string arguments, Collection<string>? argumentList)
+        {
+            // Environment variables
+            var variablesTruncated = EnvironmentVariablesScrubber.ScrubEnvironmentVariables(environmentVariables);
+            variablesTruncated = Truncate(variablesTruncated, MaxCommandLineLength, out _);
+
+            var tags = new ProcessCommandStartTags
+            {
+                EnvironmentVariables = variablesTruncated,
+                Component = "process",
```

Comment:
It's unclear why this tag cmd.component is need, could you add the reason to the PR description?

Reply:
It's not clear in the current RFC version but the `component` tag provides a better understanding of the instrumented function. The idea behind it is that if certain functions have actions against shell injection, we can adapt the backend detection accordingly. Additionally, it can assist the user in locating the part of the code that executes the command.